dapp/src/components/buySell/ApproveSwap.js

```diff
@@ -0,0 +1,264 @@
+import React, { useState, useEffect } from 'react'
+import { fbt } from 'fbt-runtime'
+import { useStoreState } from 'pullstate'
+import { useWeb3React } from '@web3-react/core'
+import ContractStore from 'stores/ContractStore'
+import analytics from 'utils/analytics'
+import withRpcProvider from 'hoc/withRpcProvider'
+import { ethers } from 'ethers'
+import withIsMobile from 'hoc/withIsMobile'
+
+const ApproveSwap = ({
+  stableCoinToApprove,
+  needsApproval,
+  selectedSwap,
+  swapMetadata,
+  onSwap,
+  allowancesLoaded,
+  onMintingError,
+  formHasErrors,
+  swappingGloballyDisabled,
+  storeTransaction,
+  storeTransactionError,
+  rpcProvider,
+  isMobile,
+}) => {
+  const [coinApproved, setCoinApproved] = useState(false)
+  const [stage, setStage] = useState(coinApproved ? 'done' : 'approve')
```

Comment:
Since line 26 always starts initial state as `false`. Line 27 initial state should always default to `approve`? Or am I not getting something?